metecho/api/models.py

```diff
@@ -694,7 +733,9 @@ def subscribable_by(self, user):  # pragma: nocover
 
     def get_absolute_url(self):
         # See src/js/utils/routes.ts
-        return f"/projects/{self.epic.project.slug}/{self.epic.slug}/{self.slug}"
+        if self.epic:
+            return f"/projects/{self.epic.project.slug}/{self.epic.slug}/{self.slug}"
+        return f"/projects/{self.project.slug}/tasks/{self.slug}"
```

Comment:
TODO: update this string once the frontend has routing in place for independent tasks

Reply:
@jerivas So this raises a question -- how do we want the user-facing URLs structured for these new epic-less tasks? Some observations:

1. Right now, our pattern for a Task is: `/projects/:projectSlug/:epicSlug/:taskSlug`
2. Ideally, we could add a new pattern without changing existing URLs (I'm not sure how important this is with the current usage of Metecho? But it's a goal, I'd think.)

In retrospect, I wish we'd have used a slightly more verbose pattern, e.g. `/projects/:projectSlug/epics/:epicSlug/tasks/:taskSlug`. But we didn't. So, options off the top of my head for this new pattern would be:

1. `/projects/:projectSlug/:taskSlug` -- This has the downside of being unable to distinguish between a Task or Epic belonging to a project. So not a great option, IMO.
2. `/projects/:projectSlug/tasks/:taskSlug` -- Allows us to leave current URLs intact, and adds a new pattern. Downside is that it reserves `tasks` as a special thing, meaning we would need to prevent users from creating an Epic with the name "Tasks".
3. Ditch the current pattern, and transition to `/projects/:projectSlug/epics/:epicSlug/tasks/:taskSlug` and `/projects/:projectSlug/tasks/:taskSlug`. This is clearer, and we could redirect from old patterns to the new ones to make the transition smoother. One minor downside is that URLs will be slightly longer?

Any thoughts -- either preferences, or different ideas?

cc @davisagli

Reply:
I lean towards option 3, I like the clarity

Reply:
@jgerigmeyer a redirect from the old path is nice, but not critical.

Reply:
@oluoluoxenfree See discussion above -- we'll want to adjust the current routes, and add a new pattern for epic-less tasks: `/projects/:projectSlug/tasks/:taskSlug`